src/core/tests/library/LegacyEnvLayerExtensions.py

```diff
@@ -579,6 +603,9 @@ def run_command_output(self, cmd, no_output=False, chk_err=True):
                     if cmd.find('sudo zypper refresh') > -1:
                         code = 6
                         output = 'Warning: There are no enabled repositories defined. | Use \'zypper addrepo\' or \'zypper modifyrepo\' commands to add or enable repositories.'
+                    elif cmd.find('sudo zypper --non-interactive update') > -1:
```

Comment:
This can be more specific to only catch the specific command instance where you want to return 8. (Use unique package details and incorporate here)

Reply:
Fixed + added a check to ensure that code 8 is returned in the UT.